### PR TITLE
[a11y] Slider should respect bold text setting

### PR DESCRIPTION
**Pull request by hannah-hyj**
Fix https://github.com/flutter/flutter/issues/147600  

internal GAR issue: b/316933135

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/test/material/slider_test.dart [hunk 1]
```diff
@@ -975,6 +1005,82 @@ void main() {
...
 
+  testWidgets('Slider value indicator respects bold text', (WidgetTester tester) async {
+      final Key sliderKey = UniqueKey();
+      double value = 0.0;
+      final List<InlineSpan> log = <InlineSpan>[];
+      final LoggingValueIndicatorShape loggingValueIndicatorShape = LoggingValueIndicatorShape(log);
+
+      Widget buildSlider({
+        bool boldText = false,
+      }) {
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:1016]:
```suggestion
      Widget buildSlider({ bool boldText = false }) {
```

File: packages/flutter/test/material/slider_test.dart [hunk 2]
```diff
@@ -49,6 +49,36 @@ class LoggingThumbShape extends SliderComponentShape {
   }
 }
 
+// A value indicator shape to log labelPainter text.
+class LoggingValueIndicatorShape extends SliderComponentShape {
+  LoggingValueIndicatorShape(this.logLabel);
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:54]: Nice!

File: packages/flutter/test/material/slider_test.dart [hunk 3]
```diff
@@ -975,6 +1005,82 @@ void main() {
...
+                return MediaQuery(
+                  data: MediaQueryData(boldText: boldText),
+                  child: Material(
+                    child: Theme(
+                      data: Theme.of(context).copyWith(
+                        sliderTheme: Theme.of(context).sliderTheme.copyWith(
+                          showValueIndicator: ShowValueIndicator.always,
+                          valueIndicatorShape: loggingValueIndicatorShape,
+                          ),
+                      ),
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:1032]:
```suggestion
                        sliderTheme: Theme.of(context).sliderTheme.copyWith(
                          showValueIndicator: ShowValueIndicator.always,
                          valueIndicatorShape: loggingValueIndicatorShape,
                        ),
                      ),
```

File: packages/flutter/test/material/slider_test.dart [hunk 4]
```diff
@@ -975,6 +1005,82 @@ void main() {
...
+      final Key sliderKey = UniqueKey();
+      double value = 0.0;
+      final List<InlineSpan> log = <InlineSpan>[];
+      final LoggingValueIndicatorShape loggingValueIndicatorShape = LoggingValueIndicatorShape(log);
+
+      Widget buildSlider({
+        bool boldText = false,
+      }) {
+        return MaterialApp(
+          theme: ThemeData(useMaterial3: false),
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:1018]: Unless the fix is specific to M2, we could be testing for M3 by default and remove this line.

**Review by TahaTesser — CHANGES_REQUESTED** [review]
Thanks for the fix!

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Comment by hangyujin** [comment]
The PR should be merging into flutter:master instead of flutter:main , reopening it as https://github.com/flutter/flutter/pull/149053, @TahaTesser can you review it again?

**Pull request closed without merging**